In [2]:
import csv
import os
import glob
import pandas as pd
import numpy as np

#data import/show

data = pd.read_csv('datalabel1.csv', index_col = [0])
data.head()


filename  (500, 501)  (501, 502)  \
0  ./20131112_P136sols/monopeaklists/20131112_P13...           0           0   
1  ./20131112_P136sols/monopeaklists/20131112_P13...           0           0   
2  ./20131112_P136sols/monopeaklists/20131112_P13...           0           0   
3  ./20131112_P136sols/monopeaklists/20131112_P13...           0           0   
4  ./20131112_P136sols/monopeaklists/20131112_P13...           0           0   

   (502, 503)  (503, 504)  (504, 505)  (505, 506)  (506, 507)  (507, 508)  \
0           0           0           0           0           0           0   
1           0           0           0           0           0           0   
2           0           0           0           0           0           0   
3           0           0           0           0           0           0   
4           0           0           0           0           0           0   

   (508, 509)      ...       (3990, 3991)  (3991, 3992)  (3992, 3993)  \
0           0      ...                  0             0             0   
1           0      ...                  0             0             0   
2           0      ...                  0             0             0   
3           0      ...                  0             0             0   
4           0      ...                  0             0             0   

   (3993, 3994)  (3994, 3995)  (3995, 3996)  (3996, 3997)  (3997, 3998)  \
0             0             0             0             0             0   
1             0             0             0             0             0   
2             0             0             0             0             0   
3             0             0             0             0             0   
4             0             0             0             0             0   

   (3998, 3999)  (3999, 4000)  
0             0             0  
1             0             0  
2             0             0  
3             0             0  
4             0             0  

[5 rows x 3501 columns]

In [6]:
reduced1 = data.loc[:, (data != 0).any(axis=0)] #with labels, no zero rows
filelabels = reduced1.iloc[:,0] # labels only 
reduced = reduced1.iloc[:,1:] # no labels

count = 0
for index, row in reduced1.iterrows():
    if row['(2705, 2706)'] == 1:
        count += 1
labels = reduced1.iloc[:,0]
str = labels.tolist()
#print(count)
#print(reduced1['(1706, 1707)'])
#reduced.shape
df1 = reduced1

### Search data for biomarkers with Gu's 

In [ ]:
df1 = reduced1
## Apodemus
for index, row in df1.iterrows():
    if row['(1293, 1294)'] == 1 and row['(1443, 1444)'] == 1 and row ['(1451, 1452)'] == 1 and row ['(1592, 1593)'] == 1 and row['(2595, 2596)'] == 1 and row['(2680, 2681)'] == 1 and row['(2695, 2696)'] == 1 and row['(2941, 2942)'] == 1:
        print(row[0])

In [ ]:
## Mammuthus
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row['(1251, 1252)'] == 1 and row ['(1518, 1519)'] == 1 and row['(1568, 1569)'] == 1 and row['(1756, 1757)'] == 1 and row['(1772, 1773)'] == 1:
        print(row[0])

In [ ]:
## Equus
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1604, 1605)'] == 1 and row['(1783, 1784)'] == 1 and row['(1182, 1183)'] == 1 and row['(1235, 1236)'] == 1 and row['(1311, 1312)'] == 1 and row['(1427, 1428)'] == 1 and row['(1501, 1502)'] == 1  and row['(1530, 1531)'] == 1 and row['(2583, 2584)'] == 1:
        print(row[0])

In [ ]:
## Coelodonta
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1604, 1605)'] == 1 and row['(1783, 1784)'] == 1 and row['(1261, 1262)'] == 1 and row['(1453, 1454)'] == 1 and row['(1473, 1474)'] == 1 and row['(1560, 1561)'] == 1 and row['(2571, 2572)'] == 1:
        print(row[0])

In [ ]:
## Bos/Bison
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1208, 1209)'] == 1 and row['(2853, 2854)'] == 1:
        print(row[0])

In [ ]:
## Cervine
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1311, 1312)'] == 1 and row['(2899, 2900)'] == 1 and row['(1550, 1551)'] == 1 and row['(2583, 2584)'] == 1:
        print(row[0])

In [ ]:
## Rangifer
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1311, 1312)'] == 1 and row['(2899, 2900)'] == 1 and row['(1580, 1581)'] == 1 and row['(2557, 2558)'] == 1:
        print(row[0])

In [ ]:
## Lepus
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1311, 1312)'] == 1 and row['(1501, 1502)'] == 1 and row['(1532, 1533)'] == 1 and row['(1608, 1609)'] == 1 and row['(2525, 2526)'] == 1:
        print(row[0])

In [ ]:
## Crocuta/Panthera
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(1560, 1561)'] == 1 and row['(2976, 2977)'] == 1 and row['(1588, 1589)'] == 1  and row['(2247, 2248)'] == 1:
        print(row[0])

In [ ]:
## Mustela
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(1560, 1561)'] == 1 and row['(2976, 2977)'] == 1 and row['(2216, 2217)'] == 1  and row['(866, 867)'] == 1 and row['(1235, 1236)'] == 1 and row['(1548, 1549)'] == 1 and row['(1680, 1681)'] == 1 and row['(2147, 2148)'] == 1:
        print(row[0])

In [ ]:
## Ursus
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(1560, 1561)'] == 1 and row['(2976, 2977)'] == 1 and row['(2216, 2217)'] == 1  and row['(836, 837)'] == 1 and row['(1690, 1691)'] == 1 and row['(1706, 1707)'] == 1 and row['(1233, 1234)'] == 1 and row['(2135, 2136)'] == 1 and row['(2163, 2164)'] == 1 and row['(2599, 2600)'] == 1:
        print(row[0])

In [ ]:
## Canid
for index, row in df1.iterrows():
    if row['(1706, 1707)'] == 1 and row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(1560, 1561)'] == 1 and row['(2976, 2977)'] == 1 and row['(2216, 2217)'] == 1  and row['(836, 837)'] == 1 and row['(1690, 1691)'] == 1 and row['(1706, 1707)'] == 1 and row['(2131, 2132)'] == 1 and row['(2613, 2614)'] == 1:
        print(row[0])

In [ ]:
##df.loc[(df['column_name'] == some_value) & df['other_column'].isin(some_values)]

In [ ]:
#check for "mammuthus" biomarkers in closely clustered samples

for index, row in reduced1.iterrows():
    if row['filename'] == './20131112_P136sols/monopeaklists/20131112_P136sols_B2.csv': #& row['(2705, 2706)'] == 1 :
        print('/20131112_P136sols/monopeaklists/20131112_P136sols_B2.csv -> sample found')
        if row['(1706, 1707)'] == 1:
            print("yes: 1706, 1707")
        if row['(2705, 2706)'] == 1:
            print("yes: 2705, 2706")
        if row['(1518, 1519)'] == 1:
            print("yes: 1518, 1519")
        if row['(1251, 1252)'] == 1:
            print("yes: 1251, 1252")
        if row['(1568, 1569)'] == 1:
            print("yes: 1568, 1569")
        if row['(1756, 1757)'] == 1:
            print("yes: 1756, 1757")
        if row['(1772, 1773)'] == 1:
            print("yes: 1772, 1773")
            
            
for index, row in reduced1.iterrows():
    if row['filename'] == './20131112_P136sols/monopeaklists/20131112_P136sols_D10.csv':
        print('./20131112_P136sols/monopeaklists/20131112_P136sols_D10.csv -> sample found')
        if row['(1706, 1707)'] == 1:
            print("yes: 1706, 1707")
        if row['(2705, 2706)'] == 1:
            print("yes: 2705, 2706")
        if row['(1518, 1519)'] == 1:
            print("yes: 1518, 1519")
        if row['(1251, 1252)'] == 1:
            print("yes: 1251, 1252")
        if row['(1568, 1569)'] == 1:
            print("yes: 1568, 1569")
        if row['(1756, 1757)'] == 1:
            print("yes: 1756, 1757")
        if row['(1772, 1773)'] == 1:
            print("yes: 1772, 1773")
            
                        
for index, row in reduced1.iterrows():
    if row['filename'] == './20131112_P140sols/monopeaklists/20131112_P140sols_I19.csv':
        print('./20131112_P140sols/monopeaklists/20131112_P140sols_I19.csv -> sample found')
        if row['(1706, 1707)'] == 1:
            print("yes: 1706, 1707")
        if row['(2705, 2706)'] == 1:
            print("yes: 2705, 2706")
        if row['(1518, 1519)'] == 1:
            print("yes: 1518, 1519")
        if row['(1251, 1252)'] == 1:
            print("yes: 1251, 1252")
        if row['(1568, 1569)'] == 1:
            print("yes: 1568, 1569")
        if row['(1756, 1757)'] == 1:
            print("yes: 1756, 1757")
        if row['(1772, 1773)'] == 1:
            print("yes: 1772, 1773")
            
for index, row in reduced1.iterrows():
    if row['filename'] == './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_L5.csv':
        print('./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_L5.csv -> sample found')
        if row['(1706, 1707)'] == 1:
            print("yes: 1706, 1707")
        if row['(2705, 2706)'] == 1:
            print("yes: 2705, 2706")
        if row['(1518, 1519)'] == 1:
            print("yes: 1518, 1519")
        if row['(1251, 1252)'] == 1:
            print("yes: 1251, 1252")
        if row['(1568, 1569)'] == 1:
            print("yes: 1568, 1569")
        if row['(1756, 1757)'] == 1:
            print("yes: 1756, 1757")
        if row['(1772, 1773)'] == 1:
            print("yes: 1772, 1773")
            
            
for index, row in reduced1.iterrows():
    if row['filename'] == './20131217_PH100solRun/monopeaklists/cals_L24.csv':
        print('./20131217_PH100solRun/monopeaklists/cals_L24.csv -> sample found')
        if row['(1706, 1707)'] == 1:
            print("yes: 1706, 1707")
        if row['(2705, 2706)'] == 1:
            print("yes: 2705, 2706")
        if row['(1518, 1519)'] == 1:
            print("yes: 1518, 1519")
        if row['(1251, 1252)'] == 1:
            print("yes: 1251, 1252")
        if row['(1568, 1569)'] == 1:
            print("yes: 1568, 1569")
        if row['(1756, 1757)'] == 1:
            print("yes: 1756, 1757")
        if row['(1772, 1773)'] == 1:
            print("yes: 1772, 1773")

### Search data for biomarkers with old

In [ ]:
df1 = reduced1
## Apodemus
for index, row in df1.iterrows():
    if row['(1293, 1294)'] == 1 and row['(1443, 1444)'] == 1 and row ['(1451, 1452)'] == 1 and row ['(1592, 1593)'] == 1 and row['(2695, 2696)'] == 1 and row['(2941, 2942)'] == 1:
        print(row[0])

In [67]:
## Mammuthus

listmam = []
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1251, 1252)'] == 1 and row['(1568, 1569)'] == 1:
        print(row[0])
        listmam.append(row[0])

./20140116_PH56SOLrun/monopeaklists/20140116_PH56SOLrun_N4.csv
./20140116_PH56SOLrun/monopeaklists/20140116_PH56SOLrun_F12.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_O1.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_O21.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_I24.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_K21.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_C18.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_A24.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_I9.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_F6.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_J21.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_J23.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_D15.csv
./20140116_PH52SOLrun/monopeaklists/20140116_PH52SOLrun_E7.csv
./20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_J6.csv
./20140115_PH80SOLrun/monopeaklists/20140115_P

In [68]:
#mamlist
lismam = []
mampd = df1.loc[df1['filename'].isin(listmam)]
print(mampd)
for i in mampd.index[:]:
    lismam.append(i) 
print(lismam)

                                               filename  (699, 700)  \
542   ./20140116_PH56SOLrun/monopeaklists/20140116_P...           0   
593   ./20140116_PH56SOLrun/monopeaklists/20140116_P...           0   
710   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
740   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
758   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
833   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
854   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
878   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
908   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
914   ./20140113_PH88SOL/monopeaklists/20140113_PH88...           0   
1105  ./20131112_P132sols/monopeaklists/20131112_P13...           0   
1338  ./20131112_P132sols/monopeaklists/20131112_P13...           0   
1356  ./20131112_P132sols/monopeaklists/20131112_P13...           0   
1377  

In [86]:
from sklearn.cluster import KMeans


#compute distances, fit data
kmeans = KMeans(n_clusters=3)
kmeans.fit(reduced)
y_kmeans = kmeans.predict(reduced)
print(y_kmeans)

[0 1 0 ... 1 1 1]


In [70]:
from sklearn.decomposition import PCA
#reduce data to 2,3 dimensions
data_reduced3 = PCA(n_components=3).fit_transform(reduced)
slice2 = data_reduced3[2953]
np.asarray(slice2)
print(data_reduced3)
data_reduced2 = PCA(n_components=2).fit_transform(reduced)
print(data_reduced3)
print(type(slice2))

[[-2.70365657 -0.42521802 -1.03048798]
 [ 0.96876586 -0.67075299 -0.91845675]
 [-3.27233784  0.21626324 -1.56281267]
 ...
 [ 0.49912673 -2.0679248   1.69560446]
 [ 4.00550603  0.09933486 -0.31804291]
 [ 3.31135306  0.0454949  -0.37167566]]
[[-2.70365657 -0.42521802 -1.03048798]
 [ 0.96876586 -0.67075299 -0.91845675]
 [-3.27233784  0.21626324 -1.56281267]
 ...
 [ 0.49912673 -2.0679248   1.69560446]
 [ 4.00550603  0.09933486 -0.31804291]
 [ 3.31135306  0.0454949  -0.37167566]]
<class 'numpy.ndarray'>


In [71]:
lismam = np.array(lismam)
comam = data_reduced3[lismam]
    
print(comam)

[[ 1.62741663 -0.90412839  2.11352524]
 [-2.97975837 -1.13605267 -0.10106213]
 [-1.0020822   2.03008988  0.20958323]
 [-1.456646    0.27135042  1.27854738]
 [ 1.24004806 -0.67588991 -1.00660315]
 [ 3.21008239 -0.22683291 -0.46732944]
 [-0.05242667  2.04248896 -2.84399973]
 [ 2.00732988  1.68640513 -2.40733409]
 [ 2.87564146  1.2797911  -2.15123968]
 [-1.57302066  1.39889682 -1.04351068]
 [ 2.38913768  1.81988306  0.68679852]
 [ 1.37738335  1.33237435 -1.81922562]
 [-0.52325962  3.00863939  0.91524173]
 [ 4.39296095  0.60739777 -1.03276176]
 [ 1.88762188 -1.9715168   1.29598705]
 [ 4.11875993 -0.48110957  0.27923227]
 [-2.86681771 -0.77058128 -0.49537388]
 [-0.56377713  3.24707896  0.65537743]
 [-2.21407769 -0.20184476 -0.86284492]
 [ 2.28482177 -0.88966864  2.44235903]
 [ 2.30487428 -2.20947136  1.60575116]
 [-2.0724143   0.50588391  0.12222239]
 [ 1.25965195  1.07227408  2.83545002]
 [ 1.41481019 -2.49948106  1.69092353]]


In [44]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=comam[:,0],
    y=comam[:,1],
    z=comam[:,2],
    mode='markers',
    text = "Mammuthus",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found with old markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

NameError: name 'comam' is not defined

In [45]:
## Equus
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1604, 1605)'] == 1 and row['(1182, 1183)'] == 1 and row['(1235, 1236)'] == 1 and row['(1311, 1312)'] == 1 and row['(1427, 1428)'] == 1 and row['(1530, 1531)'] == 1 and row['(2583, 2584)'] == 1:
        print(row[0])

In [46]:
## Coelodonta
listcoe =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1604, 1605)'] == 1 and row['(1261, 1262)'] == 1 and row['(1453, 1454)'] == 1 and row['(1473, 1474)'] == 1:
        print(row[0])
        listcoe.append(row[0])

./20131112_P136sols/monopeaklists/20131112_P136sols_B2.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_E11.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_G13.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_O12.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_K23.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_O24.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_K3.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_N19.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_A11.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_H5.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_D19.csv
./20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_M23.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_I17.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_E18.csv


In [47]:
#coelist
liscoe = []
coepd = df1.loc[df1['filename'].isin(listcoe)]
#print(coepd)
for i in coepd.index[:]:
    liscoe.append(i) 
#print(liscoe)
liscoe = np.array(liscoe)
cocoe = data_reduced3[liscoe]
print(cocoe)

[[-2.20351719  0.30294737 -0.09820481]
 [-1.28707761  3.22449799 -0.33008863]
 [-1.10963972  2.08329071 -0.59611273]
 [-1.17497165  3.57713007  1.43609384]
 [-1.22234991  2.71781245  1.8008343 ]
 [-0.99361736  3.18761513  0.57265599]
 [-0.51282112  2.97977358  1.64857897]
 [-1.24987223  3.58514843  0.19340574]
 [-0.83659801  3.89690113  0.519311  ]
 [-1.23680398  2.42833639  1.67714897]
 [-1.51657158  1.79152133  1.74950578]
 [-1.46957401  2.84350324  1.38281618]
 [-1.55210458  2.60386392 -0.33144822]
 [-0.7731999   3.45163814  2.31629919]
 [-1.45760547  3.11826156  0.66059137]]


In [48]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=cocoe[:,0],
    y=cocoe[:,1],
    z=cocoe[:,2],
    mode='markers',
    text = "Coelodonta",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found with old markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [49]:
## Bos/Bison
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1208, 1209)'] == 1 and row['(2853, 2854)'] == 1:
        print(row[0])

In [50]:
## Cervine
listcer =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1311, 1312)'] == 1 and row['(2899, 2900)'] == 1 and row['(1550, 1551)'] == 1:
        print(row[0])
        listcer.append(row[0])

./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_C10.csv


In [51]:
#ranlist
liscer = []
cerpd = df1.loc[df1['filename'].isin(listcer)]
#print(coepd)
for i in cerpd.index[:]:
    liscer.append(i) 
#print(liscoe)
liscer = np.array(liscer)
cocer = data_reduced3[liscer]
print(cocer)

[[-0.95316775  3.42458515  2.16038418]]


In [52]:
## Rangifer
listran =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row ['(2106, 2107)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1427, 1428)'] == 1 and row['(1435, 1436)'] == 1 and row['(3018, 3019)'] == 1 and row['(1311, 1312)'] == 1 and row['(2899, 2900)'] == 1 and row['(1580, 1581)'] == 1:
        print(row[0])
        listran.append(row[0])

./20131112_P132sols/monopeaklists/20131112_P132sols_O7.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_O2.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_M16.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_G9.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_G15.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_G1.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_K23.csv
./20140115_PH76SOLrun/monopeaklists/20140115_PH76SOLrun_K14.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_H15.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_A21.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_A16.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_M23.csv
./20140115_PH72SOLrun/monopeaklists/20140115_PH72SOLrun_N23.csv


In [53]:
#ranlist
lisran = []
ranpd = df1.loc[df1['filename'].isin(listran)]
#print(coepd)
for i in ranpd.index[:]:
    lisran.append(i) 
#print(liscoe)
lisran = np.array(lisran)
coran = data_reduced3[lisran]
print(coran)

[[-0.27523853  4.47049934  2.88147234]
 [-1.22244699  3.89165223  1.24572638]
 [ 2.38913762  1.81998865  0.68634383]
 [-0.51318745  5.00021171  1.99281836]
 [ 1.7352403   3.75838168  2.03007129]
 [-0.88937163  4.09855087  1.46698561]
 [ 0.3827827   3.27782128  2.8706344 ]
 [-0.66946238  3.09353748  3.15808675]
 [-0.25917503  4.64188732  2.66711598]
 [-0.74250694  4.38981949  2.80103256]
 [-0.74200118  4.49905303  2.45756686]
 [-0.59834922  4.12317412  2.32884614]
 [-0.29450267  4.45541116  3.08997339]
 [ 0.82693344 -0.63329926  1.46203536]]


In [54]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=coran[:,0],
    y=coran[:,1],
    z=coran[:,2],
    mode='markers',
    text = "Rangifer",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found with old markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [55]:
## Lepus
listlep =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1311, 1312)'] == 1 and row['(1501, 1502)'] == 1 and row['(1532, 1533)'] == 1 and row['(1608, 1609)'] == 1:
        print(row[0])
        listlep.append(row[0])

./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_I9.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_K15.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_K16.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_B2.csv
./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_I17.csv
./20131218_PH92solRun/monopeaklists/20131218_PH92solRun_C22.csv


In [56]:
#leplist
lislep = []
leppd = df1.loc[df1['filename'].isin(listlep)]
#print(coepd)
for i in leppd.index[:]:
    lislep.append(i) 
#print(liscoe)
lislep = np.array(lislep)
colep = data_reduced3[lislep]
print(colep)

[[ 2.87564149  1.27973722 -2.1510189 ]
 [ 2.30848492  0.05416137  0.01547437]
 [ 2.38913762  1.81998865  0.68634383]
 [ 3.30283045  0.87102842 -1.43463242]
 [ 3.75362994  1.69351822 -1.99587727]
 [ 3.39868957  1.01737832 -1.53400486]
 [-0.16803062  4.78530918  2.89791861]]


In [57]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=colep[:,0],
    y=colep[:,1],
    z=colep[:,2],
    mode='markers',
    text = "Lepus",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found with old markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [58]:
## Crocuta/Panthera
listcro =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(2247, 2248)'] == 1:
        print(row[0])
        listcro.append(row[0])

./20140115_PH76SOLrun/monopeaklists/20140115_PH76SOLrun_E6.csv
./20140115_PH76SOLrun/monopeaklists/20140115_PH76SOLrun_F5.csv


In [59]:
#crolist
liscro = []
cropd = df1.loc[df1['filename'].isin(listcro)]
#print(coepd)
for i in cropd.index[:]:
    liscro.append(i) 
#print(liscoe)
liscro = np.array(liscro)
cocro = data_reduced3[liscro]
print(cocro)

[[2.91101202 0.08203382 2.26456222]
 [3.02099566 1.28725073 0.31054773]]


In [60]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=cocro[:,0],
    y=cocro[:,1],
    z=cocro[:,2],
    mode='markers',
    text = "Crocuta/Panthera",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found with old markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [61]:
## Mustela
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(2216, 2217)'] == 1 and row['(1235, 1236)'] == 1 and row['(1548, 1549)'] == 1 and row['(1680, 1681)'] == 1 and row['(2147, 2148)'] == 1:
        print(row[0])

In [62]:
## Ursus
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(2216, 2217)'] == 1 and row['(1690, 1691)'] == 1 and row['(2135, 2136)'] == 1 and row['(2163, 2164)'] == 1 and row['(2599, 2600)'] == 1:
        print(row[0])

In [63]:
## Canid
listcan =[]
for index, row in df1.iterrows():
    if row['(2705, 2706)'] == 1 and row['(1832, 1833)'] == 1 and row['(1848, 1849)'] == 1 and row['(1453, 1454)'] == 1 and row['(1261, 1262)'] == 1 and row['(1459, 1460)'] == 1 and row['(2216, 2217)'] == 1 and row['(1690, 1691)'] == 1 and row['(2131, 2132)'] == 1:
        print(row[0])
        listcan.append(row[0])
print(listcan)

./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv
./20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_J15.csv
['./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv', './20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_J15.csv']


In [64]:
#canlist
liscan = []
canpd = df1.loc[df1['filename'].isin(listcan)]
#print(coepd)
for i in canpd.index[:]:
    liscan.append(i) 
#print(liscoe)
liscan = np.array(liscan)
cocan = data_reduced3[liscan]
print(cocan)

[[-1.28707761  3.22449799 -0.33008863]
 [-1.25598299  2.08200587  0.87085083]]


In [65]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Data',

    marker=dict(color=y_kmeans,size=2.7,
        opacity=0.7,
        colorscale='Rainbow',
    )
)



trace2 = go.Scatter3d(
    x=cocan[:,0],
    y=cocan[:,1],
    z=cocan[:,2],
    mode='markers',
    text = "Canid",
    name = "Known taxa",
    
    marker=dict(color = 'rgba(10, 0, 0, .8)',
     size=13,
                symbol='circle',
        opacity=0.6,
                

    )
)

dataMeans = [trace, trace2]

layout = go.Layout(
    showlegend=False,
    title='K-means, 2 clusters, samples found without novel markers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [74]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Full Dataset',

    marker=dict(color= y_kmeans, size=2,
        opacity=0.9,
       colorscale='Blues',
    )
)



trace2 = go.Scatter3d(
    x=cocan[:,0],
    y=cocan[:,1],
    z=cocan[:,2],
    mode='markers',
    text = "Canid",
    name = "Canid",
    
    marker=dict(color = 'rgba(10, 0, 110, .8)',
     size=8,
                symbol='circle',
        opacity=0.9,
                

    )
)




trace3 = go.Scatter3d(
    x=comam[:,0],
    y=comam[:,1],
    z=comam[:,2],
    mode='markers',
    text = "Mammuthus",
    name = "Mammuthus",
    
    marker=dict(color = 'rgba(252.0, 141.0, 89.0, 0.8)',
     size=8,
                symbol='cross',
        opacity=0.9,
                

    )
)




trace4 = go.Scatter3d(
    x=cocoe[:,0],
    y=cocoe[:,1],
    z=cocoe[:,2],
    mode='markers',
    text = "Coelodonta",
    name = "Coelodonta",
    
    marker=dict(color = 'rgba(255.0, 200.0, 150.0, .8)',
     size=8,
                symbol='square',
        opacity=0.9,
                

    )
)




trace5 = go.Scatter3d(
    x=coran[:,0],
    y=coran[:,1],
    z=coran[:,2],
    mode='markers',
    text = "Rangifer",
    name = "Rangifer",
    
    marker=dict(color = 'rgba(145.0, 191.0, 219.0, .8)',
     size=8,
                symbol='diamond',
        opacity=0.9,
                

    )
)



trace6 = go.Scatter3d(
    x=colep[:,0],
    y=colep[:,1],
    z=colep[:,2],
    mode='markers',
    text = "Lepus",
    name = "Lepus",
    
    marker=dict(color = 'rgba(145.0, 191.0, 0.0, .8)',
     size=8,
                symbol='square-open',
        opacity=0.9,
                

    )
)


trace7 = go.Scatter3d(
    x=cocro[:,0],
    y=cocro[:,1],
    z=cocro[:,2],
    mode='markers',
    text = "Crocuta/Panthera",
    name = "Crocuta/Panthera",
    
    marker=dict(color = 'rgba(145.0, 0.0, 0.0, .8)',
     size=8,
                symbol='circle-open',
        opacity=0.9,
                

    )
)


trace8 = go.Scatter3d(
    x=cocer[:,0],
    y=cocer[:,1],
    z=cocer[:,2],
    mode='markers',
    text = "Cervine",
    name = "Cervine",
    
    marker=dict(color = 'rgba(145.0, 102.0, 0.0, .8)',
     size=8,
                symbol='diamond-open',
        opacity=0.9,
                

    )
)


dataMeans = [trace, trace2,trace3,trace4,trace5,trace6,trace7,trace8]

layout = go.Layout(
    showlegend=True,
    title='K-means Clustering: 3 clusters; Taxa Identified without novel Biomarkermarkers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

In [78]:
from kmodes.kmodes import KModes

km = KModes(n_clusters=3, init='Huang', n_init=3, verbose=1)
clusters = km.fit_predict(reduced)
labelsmodes = km.labels_
centroids = km.cluster_centroids_

#Print the cluster centroids
print(km.cluster_centroids_)
print(km.labels_)

Init: initializing centroids
Init: initializing clusters
Starting iterations...
Run 1, iteration: 1/100, moves: 2244, cost: 662568.0
Run 1, iteration: 2/100, moves: 598, cost: 662056.0
Run 1, iteration: 3/100, moves: 98, cost: 662018.0
Run 1, iteration: 4/100, moves: 10, cost: 662018.0
Init: initializing centroids
Init: initializing clusters
Starting iterations...
Run 2, iteration: 1/100, moves: 2066, cost: 660384.0
Run 2, iteration: 2/100, moves: 376, cost: 660148.0
Run 2, iteration: 3/100, moves: 72, cost: 660123.0
Run 2, iteration: 4/100, moves: 8, cost: 660123.0
Init: initializing centroids
Init: initializing clusters
Starting iterations...
Run 3, iteration: 1/100, moves: 3806, cost: 662295.0
Run 3, iteration: 2/100, moves: 448, cost: 661648.0
Run 3, iteration: 3/100, moves: 68, cost: 661648.0
Best run was number 2
[[0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]
 [0 0 0 ... 0 0 0]]
[0 2 0 ... 2 1 1]


In [87]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Full Dataset',

    marker=dict(color= y_kmeans, size=2,
        opacity=0.9,
       colorscale='Blues',
    )
)



trace2 = go.Scatter3d(
    x=cocan[:,0],
    y=cocan[:,1],
    z=cocan[:,2],
    mode='markers',
    text = "Canid",
    name = "Canid",
    
    marker=dict(color = 'rgba(10, 0, 110, .8)',
     size=8,
                symbol='circle',
        opacity=0.9,
                

    )
)




trace3 = go.Scatter3d(
    x=comam[:,0],
    y=comam[:,1],
    z=comam[:,2],
    mode='markers',
    text = "Mammuthus",
    name = "Mammuthus",
    
    marker=dict(color = 'rgba(252.0, 141.0, 89.0, 0.8)',
     size=8,
                symbol='cross',
        opacity=0.9,
                

    )
)




trace4 = go.Scatter3d(
    x=cocoe[:,0],
    y=cocoe[:,1],
    z=cocoe[:,2],
    mode='markers',
    text = "Coelodonta",
    name = "Coelodonta",
    
    marker=dict(color = 'rgba(255.0, 200.0, 150.0, .8)',
     size=8,
                symbol='square',
        opacity=0.9,
                

    )
)




trace5 = go.Scatter3d(
    x=coran[:,0],
    y=coran[:,1],
    z=coran[:,2],
    mode='markers',
    text = "Rangifer",
    name = "Rangifer",
    
    marker=dict(color = 'rgba(145.0, 191.0, 219.0, .8)',
     size=8,
                symbol='diamond',
        opacity=0.9,
                

    )
)



trace6 = go.Scatter3d(
    x=colep[:,0],
    y=colep[:,1],
    z=colep[:,2],
    mode='markers',
    text = "Lepus",
    name = "Lepus",
    
    marker=dict(color = 'rgba(145.0, 191.0, 0.0, .8)',
     size=8,
                symbol='square-open',
        opacity=0.9,
                

    )
)


trace7 = go.Scatter3d(
    x=cocro[:,0],
    y=cocro[:,1],
    z=cocro[:,2],
    mode='markers',
    text = "Crocuta/Panthera",
    name = "Crocuta/Panthera",
    
    marker=dict(color = 'rgba(145.0, 0.0, 0.0, .8)',
     size=8,
                symbol='circle-open',
        opacity=0.9,
                

    )
)


trace8 = go.Scatter3d(
    x=cocer[:,0],
    y=cocer[:,1],
    z=cocer[:,2],
    mode='markers',
    text = "Cervine",
    name = "Cervine",
    
    marker=dict(color = 'rgba(145.0, 102.0, 0.0, .8)',
     size=8,
                symbol='diamond-open',
        opacity=0.9,
                

    )
)


dataMeans = [trace, trace2,trace3,trace4,trace5,trace6,trace7,trace8]

layout = go.Layout(
    showlegend=True,
    title='K-modes Clustering: 3 clusters; Taxa Identified without novel Biomarkermarkers'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')

As it can be seen from the figure, many taxa are dispersed around and do not localise. This figure is based on biomarkers identified previously to Gu's paper as with the new biomarkers only 1 sample was identfiable. coordinates the same but not the same file.

In [ ]:
## duplicate coordinates check

full = listlep
full.extend(listcoe)
full.extend(listmam)
full.extend(listcan)
full.extend(listcro)
full.extend(listran)
print(len(full))
print(full)

In [ ]:
#fulllist
lisfull = []
fullpd = df1.loc[df1['filename'].isin(full)]
print(fullpd)

for i in fullpd.index[:]:
    lisfull.append(i) 
#print(liscoe)
lisfull = np.array(lisfull)
cofull = data_reduced3[lisfull]
print(cofull)


In [ ]:
no_dupes = [x for n, x in enumerate(cofull) if x not in cofull[:n]]
print (no_dupes) # [[1], [2], [3], [5]]

dupes = [x for n, x in enumerate(cofull) if x in cofull[:n]]
print (dupes) # [[1], [3]]

2.38913763  1.81997382  0.68673084 mam/lep/ran
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv
./20131112_P132sols/monopeaklists/20131112_P132sols_F20.csv

-1.28707758  3.22445118 -0.3305387 can/coe
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv

2.87564149  1.27970445 -2.15111653 mam/lep
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_I9.csv
./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_I9.csv

In [13]:
#ML mammoth
mlmamlist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_N7.csv'):
        mlmamlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_G17.csv'):
        mlmamlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_C21.csv'):
        mlmamlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P140sols_H10.csv'):
        mlmamlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P140sols_I24.csv'):
        mlmamlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P140sols_L9.csv'):
        mlmamlist.append(row[0])
        
print(mlmamlist)

['./20131209_P96solsRun/monopeaklists/20131209_P96solsRun_N7.csv', './20131218_PH92solRun/monopeaklists/20131218_PH92solRun_G17.csv', './20140116_PH60SOLrun/monopeaklists/20140116_PH60SOLrun_C21.csv', './20131112_P140sols/monopeaklists/20131112_P140sols_H10.csv', './20131112_P140sols/monopeaklists/20131112_P140sols_I24.csv', './20131112_P140sols/monopeaklists/20131112_P140sols_L9.csv']


In [25]:
#mlmamlist
lismlmam = []
mlmampd = df1.loc[df1['filename'].isin(mlmamlist)]
#print(coepd)
for i in mlmampd.index[:]:
    lismlmam.append(i) 
#print(liscoe)
lismlmam = np.array(lismlmam)
comlmam = data_reduced3[lismlmam]
print(comlmam)

[[-1.32866718  2.44563248 -0.18796156]
 [-1.78668642  1.43925438 -0.16079045]
 [-1.87190208  0.39793946  0.39187263]
 [-0.84708478  2.1430092   1.30897789]
 [-1.11334134  0.89351299  1.43099102]
 [-0.93631285  2.5061577   0.8021261 ]]


In [14]:
#ML mustelid
mlmuslist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P140sols_P6.csv'):
        mlmuslist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_I18.csv'):
        mlmuslist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_G17.csv'):
        mlmuslist.append(row[0])

        
print(mlmuslist)

['./20131112_P140sols/monopeaklists/20131112_P140sols_P6.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_I18.csv', './20140116_PH52SOLrun/monopeaklists/20140116_PH52SOLrun_G17.csv']


In [26]:
#mlmuslist
lismlmus = []
mlmuspd = df1.loc[df1['filename'].isin(mlmuslist)]
#print(coepd)
for i in mlmuspd.index[:]:
    lismlmus.append(i) 
#print(liscoe)
lismlmus = np.array(lismlmus)
comlmus = data_reduced3[lismlmus]
print(comlmus)

[[-0.07918732  2.16333666  2.57952495]
 [-1.72783472  1.53062732 -1.00637202]
 [-1.0004468   2.76120966 -1.02518206]]


In [15]:
#ML lagomorph
mllaglist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_E23.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P136sols_C18.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_K5.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_L8.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_B24.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_G5.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_B5.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_G19.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_P6.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_K20.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH76SOLrun_F23.csv'):
        mllaglist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_K6.csv'):
        mllaglist.append(row[0])

        
print(mllaglist)

['./20131112_P132sols/monopeaklists/20131112_P132sols_E23.csv', './20131112_P136sols/monopeaklists/20131112_P136sols_C18.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_K5.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_L8.csv', './20131218_PH92solRun/monopeaklists/20131218_PH92solRun_B24.csv', './20131218_PH92solRun/monopeaklists/20131218_PH92solRun_G5.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_B5.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_G19.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_P6.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_K20.csv', './20140115_PH76SOLrun/monopeaklists/20140115_PH76SOLrun_F23.csv', './20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_K6.csv']


In [27]:
#mllaglist
lismllag = []
mllagpd = df1.loc[df1['filename'].isin(mllaglist)]
#print(coepd)
for i in mllagpd.index[:]:
    lismllag.append(i) 
#print(liscoe)
lismllag = np.array(lismllag)
comllag = data_reduced3[lismllag]
print(comllag)

[[-1.18272281  1.89810655 -1.41451459]
 [-1.20481081  2.36996002 -1.31444973]
 [ 0.18039588  3.1424201  -0.11473219]
 [-0.75687435  2.2597665  -1.06975221]
 [ 0.78600367  2.61530158  0.83959236]
 [ 0.05750132 -0.40310743  3.01210288]
 [ 0.91699846  1.44987234  2.63060183]
 [-0.27859218  2.17111282 -1.43904589]
 [ 1.11054649  3.37281854  0.84195082]
 [ 0.91825176  2.0216935   1.19473007]
 [-1.55199888  0.77346517  0.32403766]
 [-1.30698557  0.92262337  0.28028077]]


In [16]:
#ML hyaena
mlhyalist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_B19.csv'):
        mlhyalist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_B8.csv'):
        mlhyalist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_C13.csv'):
        mlhyalist.append(row[0])


        
print(mlhyalist)

['./20140113_PH88SOL/monopeaklists/20140113_PH88SOL_B19.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_B8.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_C13.csv']


In [28]:
#mlhyalist
lismlhya = []
mlhyapd = df1.loc[df1['filename'].isin(mlhyalist)]
#print(coepd)
for i in mlhyapd.index[:]:
    lismlhya.append(i) 
#print(liscoe)
lismlhya = np.array(lismlhya)
comlhya = data_reduced3[lismlhya]
print(comlhya)

[[-1.25183356  2.79071666  0.68186569]
 [-0.87563297  2.0456752   1.94788703]
 [-0.84296327  3.36234913  0.03782778]]


In [17]:
#ML horse
mlhorlist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_G23.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P136sols_E20.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_B5.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_D10.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_A13.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_D13.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_I22.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_N14.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_A10.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_E9.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_I24.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_P7.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_K4.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140303_PH100solRun_A13.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140303_PH100solRun_D13.csv'):
        mlhorlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140303_PH100solRun_I22.csv'):
        mlhorlist.append(row[0])

print(mlhorlist)

['./20131112_P132sols/monopeaklists/20131112_P132sols_G23.csv', './20131112_P136sols/monopeaklists/20131112_P136sols_E20.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_B5.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_D10.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_A13.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_D13.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_I22.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_N14.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_A10.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_E9.csv', './20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_I24.csv', './20140115_PH80SOLrun/monopeaklists/20140115_PH80SOLrun_P7.csv', './20140116_PH52SOLrun/monopeaklists/20140116_PH52SOLrun_K4.csv']


In [29]:
#mlhorlist
lismlhor = []
mlhorpd = df1.loc[df1['filename'].isin(mlhorlist)]
#print(coepd)
for i in mlhorpd.index[:]:
    lismlhor.append(i) 
#print(liscoe)
lismlhor = np.array(lismlhor)
comlhor = data_reduced3[lismlhor]
print(comlhor)

[[-2.31482363  0.20052974  0.32618126]
 [-0.61815278  2.22009158  2.4437809 ]
 [-0.35562086  2.051928    2.24817015]
 [-0.4352371   1.67541477  1.8024358 ]
 [-0.07259796  1.20377252  3.03955209]
 [-0.94355788  1.03963681  2.85332377]
 [-0.97617224  1.30717198  2.36953403]
 [ 0.05682925  1.65706512  3.53596702]
 [-0.65097147  0.66634592  3.19815691]
 [-0.82063309  2.99878409  1.16195423]
 [-0.25009762  3.02677233  1.37475307]
 [-0.15272038  2.73842493  2.06865324]
 [-1.7263481   1.86465342 -0.31393473]]


In [19]:
#ML fox
mlfoxlist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_D20.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_E4.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_E6.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_K14.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_K6.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_E9.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_C23.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_F3.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_D11.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_E4.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_D1.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_D1.csv'):
        mlfoxlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_O8.csv'):
        mlfoxlist.append(row[0])
        
print(mlfoxlist)

['./20131112_P132sols/monopeaklists/20131112_P132sols_D20.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_E4.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_E6.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_K14.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_K6.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_E9.csv', './20131218_PH92solRun/monopeaklists/20131218_PH92solRun_C23.csv', './20131218_PH92solRun/monopeaklists/20131218_PH92solRun_F3.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_D11.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_E4.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_D1.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_D1.csv', './20140116_PH52SOLrun/monopeaklists/20140116_PH52SOLrun_O8.csv']


In [30]:
#mlfoxlist
lismlfox = []
mlfoxpd = df1.loc[df1['filename'].isin(mlfoxlist)]
#print(coepd)
for i in mlfoxpd.index[:]:
    lismlfox.append(i) 
#print(liscoe)
lismlfox = np.array(lismlfox)
comlfox = data_reduced3[lismlfox]
print(comlfox)

[[-1.28707759e+00  3.22455030e+00 -3.30072877e-01]
 [-4.38715074e-01  3.01402760e+00 -2.57368928e-01]
 [-5.99206140e-01  1.44948703e+00  3.56656974e+00]
 [-2.86694023e-01  1.47743538e+00  3.09543372e+00]
 [-1.60990012e-03  2.82761266e+00  2.46113319e+00]
 [-1.24217351e+00  2.65985274e+00  9.95511002e-01]
 [-1.80366864e-02  4.13513065e+00  1.77755759e+00]
 [-6.35041982e-01  1.57510754e+00  2.79333916e+00]
 [ 2.11507300e-01  1.19281666e+00  3.37222915e+00]
 [-1.82525952e-01  8.21102010e-01  4.05437221e+00]
 [-9.82886608e-01  2.71128707e+00  2.36593767e+00]
 [ 2.85016865e-02  3.26276423e+00  2.48701213e+00]]


In [20]:
#ML bovine
mlbovlist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_I1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_K10.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_L23.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P132sols_P4.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_H19.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_I1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_L6.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_E11.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_G18.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_C3.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_I23.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_J13.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131218_PH92solRun_L17.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_H15.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_J2.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_J20.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_L13.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_L24.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_P24.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH72SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH72SOLrun_H11.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH72SOLrun_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH76SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH76SOLrun_I1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH76SOLrun_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_J5.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_L24.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140115_PH80SOLrun_N13.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_F18.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_G20.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_H1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_I1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_M1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_E1.csv'):
        mlbovlist.append(row[0])       
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_A1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_E1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_I1.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_J2.csv'):
        mlbovlist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_M1.csv'):
        mlbovlist.append(row[0])
 
        
print(mlbovlist)

['./20131112_P132sols/monopeaklists/20131112_P132sols_A1.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_E1.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_I1.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_K10.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_L23.csv', './20131112_P132sols/monopeaklists/20131112_P132sols_P4.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_A1.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_E1.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_H19.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_I1.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_L6.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_M1.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_A1.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_E1.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_E11.csv', './20131217_PH100solRun

In [33]:
#mlbovlist
lismlbov = []
mlbovpd = df1.loc[df1['filename'].isin(mlbovlist)]
#print(coepd)
for i in mlbovpd.index[:]:
    lismlbov.append(i) 
#print(liscoe)
lismlbov = np.array(lismlbov)
comlbov = data_reduced3[lismlbov]
print(comlbov)

[[-2.16560526  2.59233187 -0.67574335]
 [-2.27968043  2.65821836 -0.46219206]
 [-2.16736431  2.45352361 -0.17846114]
 [-2.0742367   2.26388242 -0.03245643]
 [-2.05755981  1.10612489 -0.50565577]
 [-1.80906881  1.05082532 -0.92191719]
 [-1.13791005  3.74345326  1.5977396 ]
 [-1.83100675  1.93685251 -1.05052999]
 [-0.9308022   0.38368568  0.65609005]
 [-2.27570352  2.49759066 -0.81063275]
 [-1.16655481  3.54979111  1.92789882]
 [-1.13463801  3.91987338  1.81189663]
 [-0.99855241  0.59329575  0.73448605]
 [ 0.17861635  2.18474914  3.14401334]
 [-0.92542904  0.63953968  4.0414386 ]
 [-0.54344334  4.12822368  2.21206104]
 [-1.69553999  0.33250487 -0.35974015]
 [-0.88229501  4.32373538  2.45213006]
 [-0.77925445  4.30559095  2.75783188]
 [-1.93628597 -0.96042972  1.90171697]
 [-3.12806777  0.99549806 -0.45840383]
 [-2.5257436   2.60590013 -0.1576212 ]
 [-0.26989483  3.27445824  4.47704029]
 [-2.54991798  2.14390256  0.22069776]
 [-1.78499226 -0.26390932  0.50251002]
 [-1.20171267  3.86639472

In [21]:
#ML bear
mlbealist = []

for index, row in df1.iterrows():
    if row['filename'].endswith('20131112_P136sols_C11.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131209_P96solsRun_D20.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_M6.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20131217_PH100solRun_N2.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_A14.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_H4.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140113_PH88SOL_P14.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140114_PH84SOLrun_M20.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH52SOLrun_J17.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH56SOLrun_L16.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_G11.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_M2.csv'):
        mlbealist.append(row[0])
for index, row in df1.iterrows():
    if row['filename'].endswith('20140116_PH60SOLrun_M22.csv'):
        mlbealist.append(row[0])

print(mlbealist)

['./20131112_P136sols/monopeaklists/20131112_P136sols_C11.csv', './20131209_P96solsRun/monopeaklists/20131209_P96solsRun_D20.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_M6.csv', './20131217_PH100solRun/monopeaklists/20131217_PH100solRun_N2.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_A14.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_H4.csv', './20140113_PH88SOL/monopeaklists/20140113_PH88SOL_P14.csv', './20140114_PH84SOLrun/monopeaklists/20140114_PH84SOLrun_M20.csv', './20140116_PH52SOLrun/monopeaklists/20140116_PH52SOLrun_J17.csv', './20140116_PH56SOLrun/monopeaklists/20140116_PH56SOLrun_L16.csv', './20140116_PH60SOLrun/monopeaklists/20140116_PH60SOLrun_G11.csv', './20140116_PH60SOLrun/monopeaklists/20140116_PH60SOLrun_M2.csv', './20140116_PH60SOLrun/monopeaklists/20140116_PH60SOLrun_M22.csv']


In [32]:
#mlbealist
lismlbea = []
mlbeapd = df1.loc[df1['filename'].isin(mlbealist)]
#print(coepd)
for i in mlbeapd.index[:]:
    lismlbea.append(i) 
#print(liscoe)
lismlbea = np.array(lismlbea)
comlbea = data_reduced3[lismlbea]
print(comlbea)

[[-2.22364051  1.21298014 -0.45596212]
 [-0.36424171  1.98468822  2.64016159]
 [-0.43612845  3.44965974  1.45869453]
 [-0.92600595  3.42991371  1.3136208 ]
 [ 0.41683438  3.55299118  1.06839065]
 [-0.73195096  1.29764722  2.581004  ]
 [-0.4917287   1.51852333  3.22973722]
 [ 0.00793639  1.46370182  3.18192043]
 [ 0.04653859  1.98810693  2.98040812]
 [-0.71289406  2.17647053  2.83691427]
 [-1.0467064   2.96992446 -0.1848055 ]
 [-2.01542392  1.84189129 -0.11923792]
 [-1.08665222  2.45337185  0.40481992]]


In [88]:
#visualise

import plotly.plotly as py
import plotly.graph_objs as go

import colorlover as cl


trace = go.Scatter3d(
    x=data_reduced3[:, 0],
    y=data_reduced3[:, 1],
    z=data_reduced3[:, 2],
    mode='markers',
    text = str,
    name = 'Full Dataset',

    marker=dict(color= clusters, size=2,
        opacity=0.9,
       colorscale='Blues',
    )
)



trace2 = go.Scatter3d(
    x=comlhya[:,0],
    y=comlhya[:,1],
    z=comlhya[:,2],
    mode='markers',
    text = "Hyaena",
    name = "Hyaena",
    
    marker=dict(color = 'rgba(10, 0, 110, .8)',
     size=8,
                symbol='circle',
        opacity=0.9,
                

    )
)




trace3 = go.Scatter3d(
    x=comlmam[:,0],
    y=comlmam[:,1],
    z=comlmam[:,2],
    mode='markers',
    text = "Mammuth",
    name = "Mammoth",
    
    marker=dict(color = 'rgba(252.0, 141.0, 89.0, 0.8)',
     size=8,
                symbol='cross',
        opacity=0.9,
                

    )
)




trace4 = go.Scatter3d(
    x=comlbea[:,0],
    y=comlbea[:,1],
    z=comlbea[:,2],
    mode='markers',
    text = "Bear",
    name = "Bear",
    
    marker=dict(color = 'rgba(255.0, 200.0, 150.0, .8)',
     size=8,
                symbol='square',
        opacity=0.9,
                

    )
)




trace5 = go.Scatter3d(
    x=comlbov[:,0],
    y=comlbov[:,1],
    z=comlbov[:,2],
    mode='markers',
    text = "Bovine",
    name = "Bovine",
    
    marker=dict(color = 'rgba(145.0, 191.0, 219.0, .8)',
     size=8,
                symbol='diamond',
        opacity=0.9,
                

    )
)



trace6 = go.Scatter3d(
    x=comlmus[:,0],
    y=comlmus[:,1],
    z=comlmus[:,2],
    mode='markers',
    text = "Mustelid",
    name = "Mustelid",
    
    marker=dict(color = 'rgba(145.0, 191.0, 0.0, .8)',
     size=8,
                symbol='square-open',
        opacity=0.9,
                

    )
)


trace7 = go.Scatter3d(
    x=comlhor[:,0],
    y=comlhor[:,1],
    z=comlhor[:,2],
    mode='markers',
    text = "Horse",
    name = "Horse",
    
    marker=dict(color = 'rgba(145.0, 0.0, 0.0, .8)',
     size=8,
                symbol='circle-open',
        opacity=0.9,
                

    )
)


trace8 = go.Scatter3d(
    x=comlfox[:,0],
    y=comlfox[:,1],
    z=comlfox[:,2],
    mode='markers',
    text = "Fox",
    name = "Fox",
    
    marker=dict(color = 'rgba(145.0, 102.0, 0.0, .8)',
     size=8,
                symbol='diamond-open',
        opacity=0.9,
                

    )
)

trace9 = go.Scatter3d(
    x=comllag[:,0],
    y=comllag[:,1],
    z=comllag[:,2],
    mode='markers',
    text = "Lagomorph",
    name = "Lagomorph",
    
    marker=dict(color = 'rgba(0.0, 102.0, 120.0, .8)',
     size=5,
                symbol='x',
        opacity=0.9,
                

    )
)

dataMeans = [trace, trace2,trace3,trace4,trace5,trace6,trace7,trace8, trace9]

layout = go.Layout(
    showlegend=True,
    title='K-means Clustering: 3 clusters; ML identifies taxa'
   
    
)

fig = go.Figure(data=dataMeans, layout=layout)
py.iplot(fig, filename='simple-3d-scatter')